### PR TITLE
Add required PHP extensions to composer.json

### PR DESCRIPTION
**Pull request by mhujer**
Some extensions (so far I've discoverd: `exif`, `fileinfo`, `gd`, `intl`) are necessary for tests to run, so they should be listed in the `require-dev` part of the `composer.json`

They gave me hard time during #3501 because every build takes a few minutes because of the Composer dependencies :clock1:

### CODE REVIEW

**Comment by patkar** [comment]
Are you sure, they are only needed in tests an not in general?

**Comment by mhujer** [comment]
@patkar I can't say for sure. They may be necessary only for specific modules or so.

**Comment by pjedrzejewski** [comment]
@Sylius/core-team

**Comment by michalmarcinkowski** [comment]
Sylius will not run properly without `ext-intl`, so it should be in general require. Not sure about the rest...

File: composer.json [hunk 1]
```diff
@@ -75,6 +75,10 @@
         "winzou/state-machine-bundle":          "~0.2"
     },
     "require-dev": {
+        "ext-exif":                             "*",
+        "ext-fileinfo":                         "*",
+        "ext-gd":                               "*",
+        "ext-intl":                              "*",
```
Comment (umpirsky) [inline composer.json]: Indention. :tongue:

**Comment by umpirsky** [comment]
I think this should match `Sylius\Bundle\InstallerBundle\Requirement\ExtensionsRequirements`.

**Comment by mhujer** [comment]
Thanks for the feedback, I'll update the PR later today.

File: composer.lock [hunk 2]
```diff
@@ -3264,61 +3265,6 @@
             "time": "2015-08-27 11:27:33"
         },
         {
-            "name": "mathiasverraes/money",
```
Comment (mhujer) [inline composer.lock]: I used only commands such as `composer require ext-fileinfo:*` so the unrelated changes in `composer.lock` are probably result of it being out of sync with `composer.lock`.

**Comment by mhujer** [comment]
I've just updated the PR (the commits are completely new).

I disabled the extensions and tried to install Sylius with them disabled. Then, when I got an error message, I marked the missing extension as required, enabled it and tried again. See the commit messages for more details why each extension is necessary.

**Comment by michalmarcinkowski** [comment]
Good job! :+1:

**Comment by pjedrzejewski** [comment]
Great work Martin, thank you very much! :)
